src/key_exchange/group/ristretto255.rs

```diff
@@ -63,7 +62,7 @@ impl KeGroup for Ristretto255 {
                 }
             };
 
-            if scalar != Scalar::zero() && scalar.is_canonical() {
```

Comment:
`Scalar::random()` uses `Scalar::from_bytes_mod_order_wide()` (which we used before), which should do proper clamping, so this check isn't needed. Is this correct?

Reply:
Agreed